openwisp_notifications/tests/test_api.py

```diff
@@ -671,6 +804,71 @@ def _unread_notification(notification):
                 '{view}?next={url}'.format(view=reverse('admin:login'), url=url),
             )
 
+    def test_organization_notification_setting_update(self):
+        tester = self._create_user()
+        org = Organization.objects.first()
+
+        with self.subTest('Test for current user'):
+            url = self._get_path(
+                'organization_notification_setting', self.admin.pk, org.pk
+            )
+            NotificationSetting.objects.filter(
+                user=self.admin, organization_id=org.pk
+            ).update(email=False, web=False)
+            org_setting_count = NotificationSetting.objects.filter(
+                user=self.admin, organization_id=org.pk
+            ).count()
+            response = self.client.post(url, data={'web': True, 'email': True})
+            self.assertEqual(response.status_code, 200)
+            self.assertEqual(
+                NotificationSetting.objects.filter(
+                    user=self.admin, organization_id=org.pk, email=True, web=True
+                ).count(),
+                org_setting_count,
+            )
+
+        with self.subTest('Test for non-admin user'):
+            self.client.force_login(tester)
+            url = self._get_path(
+                'organization_notification_setting',
+                self.admin.pk,
+                org.pk,
+            )
+            response = self.client.post(url, data={'web': True, 'email': True})
+            self.assertEqual(response.status_code, 403)
+
+        with self.subTest('Test with invalid data'):
+            self.client.force_login(self.admin)
+            url = self._get_path(
+                'organization_notification_setting',
+                self.admin.pk,
+                org.pk,
+            )
+            response = self.client.post(url, data={'web': 'invalid'})
+            self.assertEqual(response.status_code, 400)
+
+        with self.subTest(
+            'Update organization-level web to False while keeping one of email notification setting to true'
+        ):
+            url = self._get_path(
+                'organization_notification_setting',
+                self.admin.pk,
+                org.pk,
+            )
+
+            # Set the default type notification setting's email to True
+            NotificationSetting.objects.filter(
+                user=self.admin, organization_id=org.pk, type='default'
+            ).update(email=True)
+
+            response = self.client.post(url, data={'web': True, 'email': False})
+
+            self.assertTrue(
+                NotificationSetting.objects.filter(
+                    user=self.admin, organization_id=org.pk, type='default', email=True
+                ).exists()
+            )
+
```

Comment:
This is the scenario which I think this test emulates: 

Starting state:
```python
            NotificationSetting.objects.filter(
                user=self.admin, organization_id=org.pk, type='default'
            ).update(email=True)

```

![Screenshot from 2024-09-18 20-12-10](https://github.com/user-attachments/assets/90d9a6ac-e7ae-4402-b087-1bf9ac12972e)
state: 

Turning off email notifications for the organization: 
```python 
response = self.client.post(url, data={'web': True, 'email': False})
```

![Screenshot from 2024-09-18 20-14-40](https://github.com/user-attachments/assets/eade3a07-5b24-49b0-8ebf-af789052c394)

**Contradictory to the UI**

```python
            self.assertTrue(
                NotificationSetting.objects.filter(
                    user=self.admin, organization_id=org.pk, type='default', email=True
                ).exists()
            )
```

The UI shows that all email notifications for the organization is disabled, which is correct.